crates/noirc_abi/src/input_parser/toml.rs

```diff
@@ -55,9 +55,11 @@ fn toml_map_to_field(
     for (parameter, value) in toml_map {
         let mapped_value = match value {
             TomlTypes::String(string) => {
-                let new_value = parse_str(&string)?;
+                let new_value = parse_str(&string);
 
-                if let Some(field_element) = new_value {
+                if new_value.is_err() {
+                    InputValue::String(string)
+                } else if let Ok(Some(field_element)) = new_value {
                     InputValue::Field(field_element)
```

Comment:
We don't necessarily want a string, we may want a field element from the toml string. Previously we only let you input fields so it would be an error if the string could not be parsed as a field element.  I return `InputValue::Field(...)` if a field element is parsed as we want users to be able to still pass in hex strings to represent their field element inputs. However, I see how this is confusing. I'm going to look at moving this logic into the `parse_str` method and just returning an `InputValue` from it directly if this would make it more clear. I will add some comments as well.

Reply:
I left a clarifying comment and renamed `parse_str` to `try_str_to_field`. Let me know what you think. I felt that checking for whether the string is field element first made sense as we do want an `InputValue::Field(...)` if the user passed a hex string as input to the circuit. Then if we cannot parse a field element we assume that the user wants to input a string.